src/main/java/com/thealgorithms/scheduling/HighestResponseRatioNextScheduling.java

```diff
@@ -0,0 +1,140 @@
+package com.thealgorithms.scheduling;
+
+import java.util.Arrays;
+import java.util.Comparator;
+
+/**
+ * The HighestResponseRatioNextScheduling class implements
+ * the Highest Response Ratio Next (HRRN) scheduling algorithm.
+ * HRRN is a non-preemptive scheduling algorithm that
+ * selects the process with the highest response ratio for execution.
+ * The response ratio is calculated as (waiting time + burst time) / burst time.
+ * This algorithm aims to reduce the average waiting time
+ * and improve overall system performance by balancing short and long processes.
+ */
+public final class HighestResponseRatioNextScheduling {
+    private HighestResponseRatioNextScheduling() {
+    }
+
+    /**
+     * Calculates the Turn Around Time (TAT) for each process.
+     *
```

Comment:
Good to highlight how Turn Around Time is calculated and what is this, with some formatting.

Like a variant like this, but maybe more detailed, for better understanding:
```
     * <p>Turn Around Time is calculated as the total time a process spends 
     * in the system from arrival to completion.</p>
```

Reply:
Done!